src/debug-generator/debug-generator.c

```diff
@@ -59,12 +62,26 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
                 if (value) {
                         r = parse_boolean(value);
-                        if (r < 0)
-                                log_error("Failed to parse systemd.debug_shell= argument '%s', ignoring.", value);
-                        else
-                                arg_debug_shell = r;
-                } else
-                        arg_debug_shell = true;
+                        if (r > 0) {
+                                arg_debug_shell = strdup("y");
```

Comment:
I think it would make sense to make "arg_debug_shell" just a const string (i.e. not allocated on the heap), and then simply assign DEBUGTTY here (i.e. use the build-time variable here). [And then filter out DEBUGTTY again before writing the file as an optimization]